pilot/pkg/networking/core/v1alpha3/listener_builder.go

```diff
@@ -722,3 +722,28 @@ func buildOutboundCatchAllNetworkFilterChains(_ *ConfigGeneratorImpl,
 	chains = append(chains, &listener.FilterChain{Name: VirtualOutboundCatchAllTCPFilterChainName, Filters: filterStack})
 	return chains
 }
+
+// EvaluateListenerFilterPredicatesInternal runs through the ListenerFilterChainMatchPredicate logic
+// This is expose for testing only, and should not be used in XDS generation code
```

Comment:
```suggestion
// This is exposed for testing only, and should not be used in XDS generation code
```

Reply:
EvaluateListenerFilterPredicatesInternal should be renamed IMO. Honestly specifying `Internal` in the name is redundant. Go's treatment of camelCase and PascalCase method names already imply this

Reply:
Its used by other packages, so cannot be camelCase

Reply:
Yeah, I see that. What I meant to say is that `EvaluateListenerFilterPredicates` should be chosen over `EvaluateListenerFilterPredicatesInternal`